packages/testing/src/index.ts

```diff
@@ -79,39 +80,93 @@ export function createTestkit(
   return {
     wait: ms => new Promise(resolve => setTimeout(resolve, ms)),
     replaceSchema,
-    execute: async (operation, rawVariables = {}, initialContext = {}) => {
-      const request = {
-        headers: {},
-        method: 'POST',
-        query: '',
-        body: {
-          query: typeof operation === 'string' ? operation : print(operation),
-          variables: rawVariables,
-        },
-      };
+    execute: async (operation, variableValues = {}, initialContext = {}) => {
       const proxy = initRequest(initialContext);
-      const { operationName, query, variables } = getGraphQLParameters(request);
-
-      const r = await processRequest({
-        operationName,
-        query,
-        variables,
-        request,
-        execute: proxy.execute,
-        parse: proxy.parse,
-        validate: proxy.validate,
-        contextFactory: proxy.contextFactory,
-        schema: proxy.schema,
-      });
 
-      return (r as any).payload as ExecutionResult;
-    },
-    executeRaw: async (args: ExecutionArgs) => {
-      const proxy = initRequest(args.contextValue);
-      return await proxy.execute({
-        ...args,
-        contextValue: await proxy.contextFactory(),
+      let document: DocumentNode;
+      try {
+        document = isDocumentNode(operation) ? operation : proxy.parse(operation);
+      } catch (err) {
+        return {
+          errors: [err],
+        };
+      }
+
+      let validationErrors: ReadonlyArray<GraphQLError>;
+      try {
+        validationErrors = proxy.validate(proxy.schema, document);
+      } catch (err) {
+        return {
+          errors: [err],
+        };
+      }
+
+      if (validationErrors.length > 0) {
+        return {
+          errors: validationErrors,
+        };
+      }
+
+      const mainOperation = getOperationAST(document);
+
+      if (mainOperation == null) {
+        return {
+          errors: [new GraphQLError('Could not identify main operation.')],
+        };
+      }
+
+      let contextValue: any;
+
+      try {
+        contextValue = await proxy.contextFactory({
+          request: {
+            headers: {},
+            method: 'POST',
+            query: '',
+            body: {
+              query: print(document),
+              variables: variableValues,
+            },
+          },
+          document,
+          operation: print(document),
+          variables: variableValues,
+          ...initialContext,
+        });
+      } catch (err) {
+        return {
+          errors: [err],
```

Comment:
I think we should do instanceof GraphQLError checks here and throw the original error if it does not extends a GraphQLError instead of returning it, @dotansimha what do you think? Helix seems to convert any error into a GraphQLError

Reply:
Yes! this should make it more simple to catch unexpected errors during tests